### PR TITLE
V3.1.0

### PR DESCRIPTION
**Pull request by amitsendinblue**
- Sorting order implemented for routes (`Processes`, `emailCampaigns`, `Webhooks`, `smsCampaigns`, `transactionalEmails`, `transactionalSms`)
 - Blocked domains functionality implemented
 - Added error event in response schema of `getEmailEventReport`
 - Unnecessary fields from swagger file that were initially marked as `required`, are now marked as `optional`
 - Fix in sorting order in case of marketing events
 - Fix in `get-reports` route when passing `startDate` and `endDate`
 - Return `contact ID` in webhook if ID is used in add/remove contact
 - `getFolders` and `getFoldersLists` count issues fixed
 - Add `templateId` to response in `getEmailEventReport`

### CODE REVIEW

File: src/sib_api_v3_sdk/Client/Configuration.cs [hunk 1]
```diff
@@ -110,7 +110,7 @@ static Configuration()
         /// </summary>
         public Configuration()
         {
-            UserAgent = "sendinblue_clientAPI/v3.0.0/c#";
+            UserAgent = "sendinblue_clientAPI/v#/c#";
```
Comment (ruudy-sib) [inline src/sib_api_v3_sdk/Client/Configuration.cs]: `#` should be replaced by version number, i.e. 3.1.0

File: src/sib_api_v3_sdk/Client/Configuration.cs [hunk 2]
```diff
@@ -183,7 +183,7 @@ public Configuration()
             string tempFolderPath = null,
             string dateTimeFormat = null,
             int timeout = 100000,
-            string userAgent = "sendinblue_clientAPI/v3.0.0/c#"
+            string userAgent = "sendinblue_clientAPI/v#/c#"
```
Comment (ruudy-sib) [inline src/sib_api_v3_sdk/Client/Configuration.cs]: `#` should be replaced by version number, i.e. 3.1.0

Comment (amitsendinblue) [reply src/sib_api_v3_sdk/Client/Configuration.cs]: Its done here:
https://github.com/sendinblue/APIv3-csharp-library/blob/f0e131c8d9b956054cce49279a025badf45d9c2a/src/sib_api_v3_sdk/Client/ApiClient.cs#L180

File: src/sib_api_v3_sdk/Client/Configuration.cs (hunk 1 above)
Comment (ruudy-sib) [reply src/sib_api_v3_sdk/Client/Configuration.cs]: Being replaced globally so resolving this comment

**Review by ruudy-sib — APPROVED** [review]
LGTM

**Pull request merged**